### PR TITLE
Snap: Fix #243

### LINKED ISSUE #243: Snap: rework file selection filters
The file selection filters for the snap generator are too restrictive - they need to be adjusted to all new packages. It would make more sense to add a negative filter, which excludes what is not wanted.

### PR DESCRIPTION
**Pull request by MSoegtropIMC**
- change file filter for bin folder from positive to negative list
- further restruct file filer for lib folder (exclude coq-xyz)

### CODE REVIEW

File: linux/snap/snapcraft.yaml.in [hunk 1]
```diff
@@ -24,9 +24,9 @@ parts:
...
-      find $SNAPCRAFT_PART_INSTALL/coq-platform/@@PLATFORM_RELEASE@@/bin/ -maxdepth 1 -mindepth 1 \! \( -name 'coq*' -o -name 'clight*' -o -name 'gappa' -o -name 'eprover' -o -name 'z3_tptp' \) -exec rm -f {} \;
-      find $SNAPCRAFT_PART_INSTALL/coq-platform/@@PLATFORM_RELEASE@@/lib/ -maxdepth 1 -mindepth 1 \! \( -name 'coq*' -o -name 'stublibs' \) -exec rm -rf {} \;
+      find $SNAPCRAFT_PART_INSTALL/coq-platform/@@PLATFORM_RELEASE@@/ \( -name '*.byte.exe' -o -name '*.byte' -o -name '*.cm[aioxt]' -o -name '*.cmxa' -o -name '*.[oa]' -o -name '*.cmti' -o -name '*.glob' \) -type f -delete
+      find $SNAPCRAFT_PART_INSTALL/coq-platform/@@PLATFORM_RELEASE@@/bin/ \( -name 'ocaml*' -o -name 'camlp5*' -o -name 'mkcamlp5' -o -name 'ocpp5' -o -name 'cppo' -o -name 'safe_camlp4' -o -name 'ppxfind' -o -name 'gdk_pixbuf_mlsource3' -o -name 'lablgladecc3' \) -type f -delete
+      find $SNAPCRAFT_PART_INSTALL/coq-platform/@@PLATFORM_RELEASE@@/lib/ -maxdepth 1 -mindepth 1 \( -name 'coq' -o -name 'stublibs' \) -type d -exec rm -rf {} \;
```
Comment (gares) [inline linux/snap/snapcraft.yaml.in:29]: Didn't you forget the negation in this lase one?

Comment (gares) [reply linux/snap/snapcraft.yaml.in:29]: I mean, you must keep `coq-core` and `stublibs` for example

Comment (MSoegtropIMC) [reply linux/snap/snapcraft.yaml.in:29]: Umm yes - I forgot a negation in here ... I first wanted to do a negative list for lib as well, but then found that here positive makes more sense.

**Comment by MSoegtropIMC** [comment]
@gares : I am mildly confused by the cxms files. Coq seems to require those from e.g. coq-core, but not e.g. `/lib/elpi/elpi.cmxs`. Why is this so?

**Comment by MSoegtropIMC** [comment]
The Windows failure is a spurious network error.

**Comment by MSoegtropIMC** [comment]
@gares: thanks - I will then rework it (for the next release) to go via opam file lists and filter out by file type.

**Pull request merged**